src/bun.js/node/node_fs.zig

```diff
@@ -5277,9 +5277,30 @@ pub const NodeFS = struct {
         const fd = switch (args.file) {
             .path => brk: {
                 path = args.file.path.sliceZ(pathbuf);
+                bun.path.posixToPlatformInPlace(@constCast(path));
+
+                var is_dirfd_different = false;
+                const dirfd = if (!Environment.isWindows) args.dirfd else blk: {
+                    if (std.mem.startsWith(u8, path, "..\\")) {
+                        is_dirfd_different = true;
+                        var buffer = std.mem.zeroes([std.fs.MAX_PATH_BYTES - 1:0]u8);
```

Comment:
This is potentially a 64 KB buffer. We don’t need to zero the memory. So let’s just not.